src/App.js

```diff
@@ -1,30 +1,35 @@
 import React, { useEffect,useState } from 'react';
-function App() {
-  const [days, setDays] = useState();
-  const [hours, setHours] = useState();
-  const [minutes, setMinutes] = useState();
-  const [seconds, setSeconds] = useState();
 
-  const birthday = new Date("Dec 08,2022 16:57:00").getTime();
-  const present = new Date().getTime();
-  const [difference,setDifference] = useState(birthday - present);
+function App({ date }) {
+
+  const birthday = new Date(date.toString('YYYY-MM-dd')).getTime();
 
-  const addYear = () => {
-    setDifference(birthday - present);
-    setDays(Math.floor(difference / (1000 * 60 * 60 * 24)));
-    setHours(Math.floor((difference / (1000 * 60 * 60)) % 24));
-    setMinutes(Math.floor((difference / 1000 / 60) % 60));
-    setSeconds(Math.floor((difference / 1000) % 60));
-    console.log(difference);
-  }
+  const [difference,setDifference] = useState(birthday - new Date().getTime());
+  const [days, setDays] = useState(Math.floor(difference / (1000 * 60 * 60 * 24)));
+  const [hours, setHours] = useState(Math.floor((difference / (1000 * 60 * 60)) % 24));
+  const [minutes, setMinutes] = useState(Math.floor((difference / 1000 / 60) % 60));
+  const [seconds, setSeconds] = useState(Math.floor((difference / 1000) % 60));
+  const present = new Date().getTime();
 
   useEffect (() => {
-    const interval = setInterval(addYear, 1000);
-    // console.log(seconds);
+    const interval = setInterval(
+      () => {
+        const diff = birthday - present;
+        setDifference(diff);
+        console.log(birthday, present);
+        if ( days <= 0 && hours <= 0 && minutes <= 0 && seconds <= 0) {
+          setDifference(0);
+        }
+        setDays(Math.floor(diff / (1000 * 60 * 60 * 24)));
+        setHours(Math.floor((diff / (1000 * 60 * 60)) % 24));
+        setMinutes(Math.floor((diff / 1000 / 60) % 60));
+        setSeconds(Math.floor((diff / 1000) % 60));
+      }, 1000);
     return () => {
```

Comment:
If you have only one function inside, you can use short syntax(without brackets)